src/Matterhorn/State/Messages.hs

```diff
@@ -681,6 +686,57 @@ runNotifyCommand post mentioned = do
                                  [notified, sender, messageString] Nothing Nothing
                 return Nothing
 
+encodeToJSONstring :: A.ToJSON a => a -> String
+encodeToJSONstring a = map (chr . fromEnum) . BL.unpack $ A.encode a
+
+-- We define a notifyGetPayload for each notification version.
+notifyGetPayload :: Int -> ChatState -> Post -> Bool -> MH (Maybe String)
+
+-- Notification Version 2
+data NotificationV2 = NotificationV2
+    { version :: Int
+    , message :: Text
+    , mentioned :: String
+    , sender :: Text
+    } deriving (Show)
+
+instance A.ToJSON NotificationV2 where
+    toJSON ( NotificationV2 version message mentioned sender ) =
+        A.object [ "version"    A..= version
+                 , "message"    A..= message
+                 , "mentioned"  A..= mentioned
+                 , "sender"     A..= sender
+                 ]
+
+notifyGetPayload 2 st post mentioned = do
```

Comment:
There are a bunch of compiler warnings in this model that I'm wondering if you could address. This one I spotted myself: this function will cause a runtime exception if invoked on any payload version argument other than `2` because there is no fall-through case.

Also, a separate design thought: if the valid versions are only 1 and 2, then I think it would be easier to avoid issues with handling all possible version values exhaustively by making a new `NotificationVersion` type, e.g. `data NotificationVersion = NotifyV1 | NotifyV2`. This has the benefit that pattern-matches on this type can be checked for exhaustiveness, which is a very desirable thing to do whenever possible.

Reply:
(It also prevents anyone from mistaking this `Int` to be useful in a spot where some other non-version `Int` value is used.)

Reply:
Resolved by commit 69f9a387 .

Reply:
Resolved by commit 69f9a38 .